src/api/ledger/transactions.js

```diff
@@ -62,24 +62,45 @@ function getAccountTx(remote, address, options, marker, limit, callback) {
   });
 }
 
-function getTransactionsInternal(remote, address, options, callback) {
+function checkForLedgerGaps(remote, options, transactions) {
+  let {minLedgerVersion, maxLedgerVersion} = options;
+
+  // if we reached the limit on number of transactions, then we can shrink
+  // the required ledger range to only guarantee that there are no gaps in
+  // the range of ledgers spanned by those transactions
+  if (options.limit && transactions.length === options.limit) {
```

Comment:
Well one important use case is "get the next/previous transaction after this specified transaction". We just want to make sure that there are no gaps between the two transactions, so it's ok if there is a gap further back in the history.

> This should still detect any gaps in any of the pages though yeah?

Not sure what you mean by pages as there are no pages here, just a list of all transactions about to be returned. It is supposed to check that there are no gaps in the range of ledgers spanned by these transactions.